Tests/DatadogTests/Datadog/RUMMonitorTests.swift

```diff
@@ -1077,6 +1077,15 @@ class RUMMonitorTests: XCTestCase {
         try Datadog.deinitializeOrThrow()
     }
 
+    // MARK: - Internal attributes
+
+    func testHandlingInternalTimestampAttribute() {
+        // TODO: RUMM-990
+        // this test should mock `ApplicationScope` and assert on its received `RUMCommands`
+        // we can discuss how to do this in this PR
+        return
+    }
```

Comment:
either i need to duplicate almost every test case: one with default time and one with internal timestamp.
or i will mock `ApplicationScope` and assert `receivedCommand.time`: but this requires creating `ApplicationScope` protocol, is this fine with you @ncreated ?

if i remember well, we already have reviewing those end-to-end tests one day, maybe we can leave mocking `ApplicationScope` to that day. until then i can test internal timestamp in only some `RUMMonitor` methods.?